pkg/build/controller/build/defaults/defaults.go

```diff
@@ -78,6 +81,33 @@ func setPodLogLevelFromBuild(pod *corev1.Pod, build *buildv1.Build) error {
 	return nil
 }
 
+func (b BuildDefaults) applyPodProxyDefaults(pod *corev1.Pod, isCustomBuild bool) {
+	allContainers := make([]*corev1.Container, 0, len(pod.Spec.Containers)+len(pod.Spec.InitContainers))
```

Comment:
you don't need to pre-allocate this.

`allContainers:=[]*corev1.Container{}` should work fine.  append creates a new slice every time.

Reply:
updated